fix: Fixed input for ICAO on EFB dashboard to avoid key press leaks to sim

frankkopp opened the pull request:
Fixes #6524

## Summary of Changes
Changed input component on EFB dashboard weather widget for ICAO input. 

- Used SimpleInput instead to prevent leaking key presses to the sim. 
- Added input box around ICAO in EFB dashboard weather widget to signifying that an input can be made


## Screenshots (if necessary)
See issue https://github.com/flybywiresim/a32nx/issues/6524

![image](https://user-images.githubusercontent.com/16833201/149152586-a4d10438-44de-4276-9669-da89623f7862.png)

<!-- You may optionally provide your discord username, so that we may contact you directly about the issue. -->
Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions
Start flight anywhere
Turn on EFB and enter an ICAO (or any value) into the airport ICAO fields ("----").
Make sure that no input is leaked to the sim. 
Examples: 
"L" in the sim should trigger lights on/off - should not trigger it while in ICAO input field
"O" in the sim should trigger Strobes on/off - should not trigger it while in ICAO input field

Test flight plan import and any other related option to see if you can break it. 

<!-- DO NOT DELETE THIS -->
## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

ErickSharp commented on src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:
```diff
@@ -119,11 +120,11 @@ const WeatherWidget = (props: WeatherWidgetProps) => {
+                                    className="text-center w-24 ml-4 text-2xl font-medium uppercase"
                                     placeholder={props.icao}
-                                    onChange={handleIcao}
+                                    onChange={(event) => handleIcao(event)}
```
```suggestion
                                    onChange={(value) => handleIcao(value)}
```
There’s no event, it would make more sense to use value instead

ErickSharp reviewed (approved):
😁

mico975 commented:
Quality Assurance Trainee Report

Discord          : mico#3145
Object of testing: #6525 
Tier of Testing  : 1
Date             : 13/01/2022

Testing Process: 
Spawned at KFBW

Inputed some ICAO codes with METAR stations and got correct weather info (it's insanely quick :-) )
Inputed some ICAO codes without METAR stations and got 0 everywhere - maybe it could display "/" since there is no data? Or "METAR not available" when the whole report will be implemented.
Inputed some non existing ICAO codes and also got 0 everywhere

Restarted flypad and all fields cleared to ----
Requested Simbrief - got correct weather
Overwrote ICAOs - got correct weather at other stations

Restarted flypad and all fields cleared to ----
Inputed two valid ICAOs and then requested Simbrief - it does not overwrite the previously inputted ICAO codes. If it's intended this way then it's good, but I would maybe suggest that Simbrief request overwrites any previously manually inputed ICAO codes?

Again restarted flypad and all fields cleared to ----.
Inputed two valid ICAO codes, got correct weather, changed flypad page and ICAO codes were gone after coming back to dashboard page.
Requested Simbrief again, overwrote ICAO codes, changed flypad page and on return to dashboard page weather was there for  Simbrief imported ICAO codes.

Checked for any input leaks to the sim and there were none, looks good.

Negatives:       
None, just some suggestions already written in testing process.

Testing Results: 
Passed

Conclusions: 
Inputs don't leak to the sim - works great.
Probably a bit more extensive test than required but I managed to test the thing thoroughly :)

Media:       
![image](https://user-images.githubusercontent.com/78908672/149368163-0f95461f-d869-4f71-a1f4-1889cfc798c9.png)
Simbrief import works
![image](https://user-images.githubusercontent.com/78908672/149368208-96d882eb-1bb1-4dbf-8fad-444bdd48baed.png)
LJLJ has METAR station, while LJPT does not.

frankkopp commented:
Thanks for your test!!

> Inputed some ICAO codes with METAR stations and got correct weather info (it's insanely quick :-) ) Inputed some ICAO codes without METAR stations and got 0 everywhere - maybe it could display "/" since there is no data? Or "METAR not available" when the whole report will be implemented. Inputed some non existing ICAO codes and also got 0 everywhere

Good idea. I would do this in a separate PR though. 
 
> Restarted flypad and all fields cleared to ---- Inputed two valid ICAOs and then requested Simbrief - it does not overwrite the previously inputted ICAO codes. If it's intended this way then it's good, but I would maybe suggest that Simbrief request overwrites any previously manually inputed ICAO codes?

You can and need to delete the manually entered values and the simbrief value surface automatically. I agree that simbrief should overwrite this when imported. Again separate PR. 

> Again restarted flypad and all fields cleared to ----. Inputed two valid ICAO codes, got correct weather, changed flypad page and ICAO codes were gone after coming back to dashboard page. Requested Simbrief again, overwrote ICAO codes, changed flypad page and on return to dashboard page weather was there for Simbrief imported ICAO codes.

Expected behavior

I plan to add the suggested changes to the other PR I already opened: 
https://github.com/flybywiresim/a32nx/pull/6529

ewold commented:
Quality Assurance Trainee Report

Discord: E-wold#8195
Object of testing: #6525
Tier of Testing: 1
Date: 17/01/2022
Testing Process: Used the weather ICAO Field and checked if nothing was toggled in the plane whilest typing
Negatives: N/A
Testing Results: Passed

Conclusions: Works as expected
![lebbpad](https://user-images.githubusercontent.com/15085769/149752078-8642348e-4c19-4b35-af26-5e342480ddeb.PNG)
